DEVDOCS-6233 | Content Refresh - "About Our APIs" (#951)
bigcommerce/docs: docs/b2b-edition/about/index.mdx

# About Our APIs

B2B Edition's APIs allow you to retrieve and manage B2B resources, such as Company accounts and users, sales quotes, and invoice payments. In conjunction with other [BigCommerce APIs](https://developer.bigcommerce.com/docs/start/about), you can build integrations and headless solutions for stores using B2B Edition.

> This article covers general information about B2B Edition's APIs and how to use them. See our [list of B2B Edition APIs](https://developer.bigcommerce.com/b2b-edition/apis) for endpoint-specific requirements, considerations, and use cases.

Author: ```suggestion
This article covers general information about B2B Edition's APIs and how to use them. See our [list of B2B Edition APIs](/b2b-edition/apis) for endpoint-specific requirements, considerations, and use cases.
```

## Available APIs

B2B Edition has different APIs for managing store and Company data, each geared toward specific environments and experiences. See the descriptions below to identify the APIs that best fit your integration.

### Rest Management V3 APIs

**Base URL** &mdash; `https://api-b2b.bigcommerce.com/api/v3/io/`

The Rest Management V3 APIs allow you to manage B2B store data and settings from a server-to-server context. Use these APIs to perform backend actions as a store administrator.

Example use cases include:

* Creating new Company accounts and setting up Company account hierarchies  
* Submitting sales quotes to buyers  
* Generating payable invoices for purchase orders  
* Retrieving B2B order information  
* Assigning user roles to B2B buyers

> 

Author: ```suggestion

Unlike the other BigCommerce REST Management APIs, the B2B Rest Management V3 APIs do not include store-specific data in the request path, such as the store hash. For example, the [BigCommerce Get an Order](/docs/rest-management/orders#get-an-order) endpoint uses a path of `https://api.bigcommerce.com/stores/{store_hash}/v2/orders/{order_id}`, while the [B2B equivalent](/b2b-edition/apis/rest-management/order#get-an-order) uses `https://api-b2b.bigcommerce.com/api/v3/io/orders/{bcOrderId}'.

```

Reviewer: "Unlike the other BigCommerce REST Management APIs, the B2B Rest Management V2/V3 APIs do not include store-specific data in the request path, such as the store hash." In each case, the store specificity is stored in the token itself. Each respective authToken carries the specific store data from which it was created. (In your own words) @bc-Vince

### GraphQL Storefront APIs

**Base URL** &mdash; `https://api-b2b.bigcommerce.com/graphql` 

B2B Edition's GraphQL Storefront APIs include queries and mutations for managing records like Company account information, sales quotes, and invoices in native and headless storefronts. It is the recommended environment if you are building storefront integrations for B2B Edition's [Buyer Portal experience](https://support.bigcommerce.com/s/article/B2B-Edition-Buyer-Functions).

Example use cases include:

* Adding new Company users  
* Taking sales quotes to checkout  
* Curating and approving Shopping Lists  
* Starting or ending Super Admin Masquerade

Our [GraphQL Playground](https://api-b2b.bigcommerce.com/graphql/playground) includes documentation and schemas for B2B GraphQL API requests, as well as an environment for testing responses.

### REST Storefront APIs

**Base URL** &mdash; `https://api-b2b.bigcommerce.com/api/v2/`**\***

> The REST Storefront APIs also allow you to manage B2B records from the context of a specific Company user or Super Admin. It is used by the [legacy Stencil storefront experience](https://developer.bigcommerce.com/b2b-edition/docs/stencil), but the requests are also supported in the Buyer Portal experience. However, we recommend using the GraphQL Storefront API for storefront customization.

Author: ```suggestion
The REST Storefront APIs also allow you to manage B2B records from the context of a specific Company user or Super Admin. It is used by the [legacy Stencil storefront experience](/b2b-edition/docs/stencil), but the requests are also supported in the Buyer Portal experience. However, we recommend using the GraphQL Storefront API for storefront customization.
```

**\*** Most REST Storefront API endpoints use `https://api-b2b.bigcommerce.com/api/v2/`, but the [Get a Storefront authToken for a Specific Customer](/b2b-edition/apis/rest-storefront/authentication#get-a-storefront-authtoken-for-a-specific-customer) endpoint uses `https://api-b2b.bigcommerce.com/api/io/`.

### REST Management V2 APIs

**Base URL** &mdash; `https://api-b2b.bigcommerce.com/api/v2/io/`

> The REST Management V2 APIs include legacy server-to-server endpoints for managing store information and processes. 

Author: ```suggestion
The REST Management V2 APIs include legacy server-to-server endpoints for managing store information and processes. Like V3 APIs, the V2 APIs do not include store-specific data in the request path.
```

The V2 API is deprecated, and it does not support many of B2B Edition's newer features. We recommend building or upgrading your integration to use the V3 Management APIs.

## API authentication and context

Requests to the B2B Edition REST Management V3 API are always in the context of the server. To authenticate V3 API requests, generate a server to server authToken via one of the following methods:

* Create an API account from [B2B Edition's settings](https://support.bigcommerce.com/s/article/B2B-Edition-Settings)  
* Generate a token via the [Create a Server to Server Token](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/authentication#get-a-server-to-server-token) endpoint

GraphQL and REST Storefront API authentication tokens are always in the context of a specific storefront user. As a result, you do not need separate tokens for authenticating requests in client and server contexts, unlike the [BigCommerce GraphQL Storefront API](https://developer.bigcommerce.com/docs/start/authentication/graphql-storefront).

> There are several methods for generating storefront authTokens, each with distinct use cases. See [Authentication for hosted storefronts](https://developer.bigcommerce.com/b2b-edition/docs/authentication) to learn more.

Author: ```suggestion
There are several methods for generating storefront authTokens, each with distinct use cases. See [Authentication for hosted storefronts](/b2b-edition/docs/authentication) to learn more.
```

## Available B2B Resources

B2B Edition's APIs allow you to manage the following major resources:

| Resource | Description | Server to server endpoints? | Storefront endpoints, queries, and mutations? |
| :---- | :---- | :---- | :---- |

> | Addresses | Gather Company address information and add new addresses | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/address) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/company/addresses) |

Author: ```suggestion
| Addresses | Gather Company address information and add new addresses | [Yes](/b2b-edition/apis/rest-management/address) | [Yes](/b2b-edition/apis/rest-storefront/company/addresses) |
```

> | Channels | Obtain store channel details | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/channel) | No |

Author: ```suggestion
| Channels | Obtain store channel details | [Yes](/b2b-edition/apis/rest-management/channel) | No |
```

| Companies | Create and manage Company accounts. Includes sub-resources for configuring account hierarchy and managing Company user roles and permissions. | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/company) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/company) |
| Invoice Management | Generate invoices for purchase orders and log external payments | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/invoice-management) | No |

> | Orders | Get order information and assign historical orders to Company accounts | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/order) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/orders) |

Author: ```suggestion
| Orders | Get order information and assign historical orders to Company accounts | [Yes](/b2b-edition/apis/rest-management/order) | [Yes](/b2b-edition/apis/rest-storefront/orders) |
```

> | Payments | View and update Company payment methods, available credit, and net terms | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/payment) | No |

Author: ```suggestion
| Payments | View and update Company payment methods, available credit, and net terms | [Yes](/b2b-edition/apis/rest-management/payment) | No |
```

> | Quotes | Create and process sales quotes | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/quote) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/request-for-quote) |

Author: ```suggestion
| Quotes | Create and process sales quotes | [Yes](/b2b-edition/apis/rest-management/quote) | [Yes](/b2b-edition/apis/rest-storefront/request-for-quote) |
```

| Sales Staff | Assign your backend sales representatives to Company accounts | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/sales-staff) | No |

> | Shopping Lists | Build and manage Shopping Lists for repeat purchases | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/shopping-list) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/shopping-list) |

Author: ```suggestion
| Shopping Lists | Build and manage Shopping Lists for repeat purchases | [Yes](/b2b-edition/apis/rest-management/shopping-list) | [Yes](/b2b-edition/apis/rest-storefront/shopping-list) |
```

| Store Settings | Retrieve basic information from a storefront context, such as available currencies and order statuses | No | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/store) |
| Super Admins | Assign your frontend sales representatives to Company accounts and initiate or end a Masquerade session | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/super-admin) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/sales-rep) |

> | Users | Add users to Company accounts and assign roles to specify Buyer Portal access permissions | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/user) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/company/user) |

Author: ```suggestion
| Users | Add users to Company accounts and assign roles | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/user) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/company/user) |
```

> | Users | Add users to Company accounts and assign roles to specify Buyer Portal access permissions | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/user) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/company/user) |

Author: ```suggestion
| Users | Add users to Company accounts and assign roles to specify Buyer Portal access permissions | [Yes](/b2b-edition/apis/rest-management/user) | [Yes](/b2b-edition/apis/rest-storefront/company/user) |
```

> | Users | Add users to Company accounts and assign roles to specify Buyer Portal access permissions | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/user) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/company/user) |

Author: ```suggestion
| Users | Add users to Company accounts and assign roles | [Yes](/b2b-edition/apis/rest-management/user) | [Yes](/b2b-edition/apis/rest-storefront/company/user) |
```

> | Users | Add users to Company accounts and assign roles to specify Buyer Portal access permissions | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/user) | [Yes](https://developer.bigcommerce.com/b2b-edition/apis/rest-storefront/company/user) |

Author: ```suggestion
| Users | Add users to Company accounts and assign roles | [Yes](/b2b-edition/apis/rest-management/user) | [Yes](/b2b-edition/apis/rest-storefront/company/user) |
```

Like other BigCommerce APIs, B2B Edition's APIs accept requests in JSON. You must encode requests in the UTF-8 character set to ensure valid results.

See the following sections for information on the content and structure of B2B Edition API requests and their responses. 

### Requests

#### Request headers

| Header | Expected value or type | Description | Example |
| :---- | :---- | :---- | :---- |
| `authToken` | String | The authToken that validates requests to the REST Management V3 API | \- |
| `Authorization` | String | The authToken that validates requests to the REST Storefront and GraphQL Storefront APIs | \- |
| `Content-Type` | MIME types | The MIME type of the request body. The API uses this value to validate and parse the request. For more information, see [HTTP Docs: Content-Type Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type). | `application/json` |
| `Accept` | MIME types | The MIME type format that indicates which response type the request expects. For more information, see [HTTP Docs: Accept Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept). | `application/json` |

#### Request structure

> B2B Edition REST Management and Storefront APIs use the standard JSON request format of an object containing a set of key-value pairs. For example, this is what the body of a simplified request to the server to server [Create a Quote](https://developer.bigcommerce.com/b2b-edition/apis/rest-management/quote/quotes#create-a-quote) endpoint would look like:

Author: ```suggestion
B2B Edition REST Management and Storefront APIs use the standard JSON request format of an object containing a set of key-value pairs. For example, this is what the body of a simplified request to the server to server [Create a Quote](/b2b-edition/apis/rest-management/quote/quotes#create-a-quote) endpoint would look like:
```

| `Access-Control-Expose-Headers` | One or more response headers | Specifies the response headers that can be exposed for CORS scripts | [HTTP Docs: Access-Control-Expose-Headers Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Expose-Headers) | `X-Request-ID,Content-Length` |
| `Set-Cookie` | A set of key-value pairs related to the cookie | The cookie sent from the server so that the user can send it back at a later time | [HTTP Docs: Set-Cookie Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Set-Cookie) | `csrftoken={token}; expires=Wed, 20 May 2026 14:50:23 GMT; Max-Age=31449600; Path=/; SameSite=Lax` |
| `Access-Control-Allow-Origin` | One or more URLs An asterisk \* for wildcard values null for no origin support | Specifies the supported origins for code requesting to access the response | [HTTP Docs: Access-Control-Allow-Origin Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Allow-Origin) | `*` |

> | `Access-Control-Allow-Headers`  | One or more request headers | Specifies in a preflight request which headers can be used in an actual request | [HTTP Docs: Access-Control-Allow-Headers Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Allow-Headers) | `authToken, Content-Type` |

Author: ```suggestion
| `Access-Control-Allow-Headers`  | One or more request headers | Specifies in a pre-flight request which headers can be used in an actual request | [HTTP Docs: Access-Control-Allow-Headers Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Allow-Headers) | `authToken, Content-Type` |
```

> | `Access-Control-Allow-Methods` | One or more request methods | Specifies in a preflight request which methods can be used in an actual request | [HTTP Docs: Access-Control-Allow-Methods Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Allow-Methods) | `GET, POST, PUT, OPTIONS, DELETE, PATCH` |

Author: ```suggestion
| `Access-Control-Allow-Methods` | One or more request methods | Specifies in a pre-flight request which methods can be used in an actual request | [HTTP Docs: Access-Control-Allow-Methods Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Allow-Methods) | `GET, POST, PUT, OPTIONS, DELETE, PATCH` |
```

> | `Access-Control-Max-Age` | Time interval, in seconds | Specifies how long that a preflight request's results can be cached | [HTTP Docs: Access-Control-Max-Age Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Max-Age) | `1728000` |

Author: ```suggestion
| `Access-Control-Max-Age` | Time interval, in seconds | Specifies how long that a pre-flight request's results can be cached | [HTTP Docs: Access-Control-Max-Age Header (MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Access-Control-Max-Age) | `1728000` |
```

    "phoneNumber": "9876543210",
    "assignedCompanies": [
      {
        "id": 123456,
        "bcId": "15",
        "companyName": "Great Buys, Inc",
        "bcGroupName": "Great Buys",
        "assignedAt": 1721835018
      }
    ]
  }
}
```

Storefront GraphQL responses contain all the fields specified in the original query or mutation. Using the [example from earlier](#request-structure), this is what the response looks like:

```graphql filename="Example GraphQL response: currencies" showLineNumbers copy
{
    "data": {
        "currencies": {
            "currencies": [
                {
                    "currency_code": "USD",
                    "currency_exchange_rate": "1.0000000000",
                    "token": "$",
                    "decimal_token": ".",
                    "decimal_places": 2,
                    "token_location": "left",
                    "thousands_token": ","
                }
            ],
            "channelCurrencies": {
                "channel_id": 1,
                "enabled_currencies": [
                    "USD"
                ],
                "default_currency": "USD"
            }
        }
    }
}
```

## Resources

> * [About BigCommerce APIs](https://developer.bigcommerce.com/docs/start/about)  

Author: ```suggestion
* [About BigCommerce APIs](/docs/start/about)  
```

> * [B2B Edition APIs](https://developer.bigcommerce.com/b2b-edition/apis)  

Author: ```suggestion
* [B2B Edition APIs](/b2b-edition/apis)  
```

> * [Best Practices](https://developer.bigcommerce.com/b2b-edition/docs/best-practices)  

Author: ```suggestion
* [Best Practices](/b2b-edition/docs/best-practices)  
```

> * [Developer Community](https://developer.bigcommerce.com/community)  

Author: ```suggestion
* [Developer Community](/community)  
```
